### PR TITLE
Converted normalizers to be estimators

### PR DESCRIPTION
**Pull request by Zruty0**
Normalizers (except SupervisedBinning) become estimators.

### CODE REVIEW

File: test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        public void NormalizerWorkout()
+        {
+            string dataPath = GetDataPath("iris.txt");
+
+            var loader = new TextLoader(Env, new TextLoader.Arguments
+            {
+                Column = new[]{
+                    new TextLoader.Column("float1", DataKind.R4, 1),
+                    new TextLoader.Column("float4", DataKind.R4, new[]{new TextLoader.Range(1, 4) }),
+                    new TextLoader.Column("float4", DataKind.R4, new[]{new TextLoader.Range(1, 4) }),
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs]: Is it on purpose to have two float4?

Comment (Zruty0) [reply test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs]: duh...

---
In reply to: [214495561](https://github.com/dotnet/machinelearning/pull/797#discussion_r214495561) [](ancestors = 214495561)

File: src/Microsoft.ML.Data/Transforms/Normalizer.cs [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+            Binning = 3
+        }
+
+        public abstract class ColumnBase
+        {
+            public readonly string Input;
+            public readonly string Output;
+            public readonly long MaxTrainingExamples;
+
+            protected ColumnBase(string input, string output, long maxTrainingExamples)
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/Normalizer.cs]: >protected [](start = 12, length = 9)

Is there any reason not to make this and all these new public abstract classes you're creating subclassable by literally anyone? I don't see so, at least in the code written here. So how about a nice `private protected` to ease your conscience? :)

File: src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs [hunk 3]
```diff
@@ -13,22 +13,21 @@
 using Microsoft.ML.Runtime.Model.Pfa;
 using Microsoft.ML.Runtime.Internal.Internallearn;
 using Newtonsoft.Json.Linq;
+using Microsoft.ML.Runtime.EntryPoints;
+using System.Linq;
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs]: I like how we start with system namespaces, then we get into Microsoft namespaces, then towards the end we go back to system. It sort of reminds me of one of those stories where the protagonist starts at home, goes off to the big city, then comes right back where they started after many adventures, but everything's a bit different. So please keep this.

File: src/Microsoft.ML.Data/Transforms/NormalizeUtils.cs [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+[assembly: LoadableClass(typeof(void), typeof(Normalize), null, typeof(SignatureEntryPointModule), "Normalize")]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// Signature for a repository based loader of a IColumnFunction
+    /// </summary>
+    public delegate void SignatureLoadColumnFunction(ModelLoadContext ctx, IHost host, ColumnType typeSrc);
+
+    public interface IColumnFunctionBuilder
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/NormalizeUtils.cs]: >IColumnFunctionBuilder [](start = 21, length = 22)

I wonder if we can make all these interfaces internal. I know the idea was "hey we'll eventually have a bunch of transforms use this stuff" but as it is. They just sort of muddy the waters.

File: src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs [hunk 5]
```diff
@@ -13,22 +13,21 @@
...
+[assembly: LoadableClass(NormalizeTransform.LogMeanVarNormalizerSummary, typeof(IDataTransform), typeof(NormalizeTransform), typeof(NormalizeTransform.LogMeanVarArguments), typeof(SignatureDataTransform),
     NormalizeTransform.LogMeanVarNormalizerUserName, "LogMeanVarNormalizer", NormalizeTransform.LogMeanVarNormalizerShortName, "LogNormalNormalizer", "LogNormal")]
 
-[assembly: LoadableClass(NormalizeTransform.BinNormalizerSummary, typeof(NormalizeTransform), typeof(NormalizeTransform.BinArguments), typeof(SignatureDataTransform),
+[assembly: LoadableClass(NormalizeTransform.BinNormalizerSummary, typeof(IDataTransform), typeof(NormalizeTransform), typeof(NormalizeTransform.BinArguments), typeof(SignatureDataTransform),
     NormalizeTransform.BinNormalizerUserName, "BinNormalizer", NormalizeTransform.BinNormalizerShortName)]
 
-[assembly: LoadableClass(NormalizeTransform.SupervisedBinNormalizerSummary, typeof(NormalizeTransform), typeof(NormalizeTransform.SupervisedBinArguments), typeof(SignatureDataTransform),
-    NormalizeTransform.SupervisedBinNormalizerUserName, "SupervisedBinNormalizer", NormalizeTransform.SupervisedBinNormalizerShortName)]
-
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs]: Just curious, why is this going away?

**Comment by sfilipi** [comment]
>         /// A helper method to create MinMaxNormalizer transform for public facing API.

do we want to replace 'transform' with 'estimator'?

---
Refers to: src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs:250 in c4cd6b7. [](commit_id = c4cd6b73ca2ba6635a45bc43accbd879dcc35a8d, deletion_comment = False)

**Pull request merged**